src/NHibernate/Cache/StandardQueryCache.cs

```diff
@@ -184,16 +209,22 @@ public IList Get(QueryKey key, ICacheAssembler[] returnTypes, bool isNaturalKeyL
 				if (queryParameters[i].NaturalKeyLookup && result.Count == 0)
 					continue;
 
+				var key = keys[i];
 				cached[i] = true;
-				cachedKeys.Add(keys[i]);
-				cachedResults.Add(GetCacheableResult(returnTypes[i], session, result, ts));
+				cachedKeys.Add(key);
+				cachedResults.Add(GetCacheableResult(returnTypes[i], session, result, ts, GetAutoDiscoveredAliases(key)));
 			}
 
 			_cache.PutMany(cachedKeys.ToArray(), cachedResults.ToArray());
 
 			return cached;
 		}
 
+		private static string[] GetAutoDiscoveredAliases(QueryKey key)
+		{
+			return key.ResultTransformer?.AutoDiscoverTypes == true ? key.ResultTransformer.AutoDiscoveredAliases : null;
```

Comment:
Why not simply?
```suggestion
			return key.ResultTransformer?.AutoDiscoveredAliases;
```

Reply:
Just code changes leading to sub-optimal code without me realizing.